File: LOC.c
#include<stdio.h>
int main(){
  long n, q,l,r,c;
  char p;
  char arr[100000][20];
  scanf("%ld",&n );
  for(long i=0;i<n;i++)
     scanf("%s",&arr[i][20]);
  scanf("%ld",&q );
  while(q--){
    long x=0;
    scanf("%ld%ld%ld",&l,&r,&c);
    scanf(" %c",&p);
    while(l<=r){
      if(arr[l++][c-1]==p)
        x++;
    }
    printf("%ld\n",x);
  }
}

File: LOC.java

import java.util.Scanner;
class LOC
{
  static Scanner in=new Scanner(System.in);
  public static void main(String[] args){
    int n=in.nextInt();

    String arr[]=new String[n];

    for(int i=0;i<n;i++){
      arr[i]=in.nextLine();
    }

    int q=in.nextInt();


    for(int i=0;i<q;i++){
       int x=0;
       int l=in.nextInt()-1;
       int r=in.nextInt()-1;
       int c=in.nextInt()-1;
       String p=in.nextLine();
       char t=p.charAt(0);
       for(int j=l;j<=r;j++){
         if((arr[j].charAt(c))==t)
           x++;
       }


    System.out.print(x);
    System.out.print("\n");
}

    }
  }
